lib/rules/no-whitespace-before-property.js

```diff
@@ -25,14 +61,20 @@ module.exports = function(context) {
             var prop = node.property;
 
             if (astUtils.isTokenOnSameLine(obj, prop)) {
-                if (sourceCode.isSpaceBetweenTokens(obj, prop)) {
-                    context.report({
-                        node: node,
-                        message: "Unexpected whitespace before property '{{ propName }}'.",
-                        data: {
-                            propName: prop.name
-                        }
-                    });
+                var punc = sourceCode.getTokenBefore(prop);
+
+                if (punc.type !== "Punctuator") {
+                    return;
+                }
+
+                if (punc.value === ".") {
+                    if (sourceCode.isSpaceBetweenTokens(obj, prop)) {
+                        reportError(node, prop);
+                    }
+                } else if (punc.value === "[") {
+                    if (sourceCode.isSpaceBetweenTokens(obj, punc)) {
```

Comment:
Ah, I see, `prop` vs `punc`. I had to read that many times to see the difference.